packages/ckeditor5-editor-classic/src/classiceditorui.ts

```diff
@@ -187,6 +192,61 @@ export default class ClassicEditorUI extends EditorUI {
 		} );
 	}
 
+	/**
+	 * Provides an integration between the sticky toolbar and {@link module:ui/panel/balloon/contextualballoon contextual balloon plugin}.
+	 * It allows the contextual balloon to consider the height of the
+	 * {@link module:editor-classic/classiceditoruiview~ClassicEditorUIView#stickyPanel}. It prevents the balloon from overlapping
+	 * the sticky toolbar by adjusting the balloon's position using viewport offset configuration.
+	 */
+	private _initContextualBalloonIntegration(): void {
+		if ( !this.editor.plugins.has( 'ContextualBalloon' ) ) {
+			return;
+		}
+
+		const { stickyPanel } = this.view;
+		const contextualBalloon = this.editor.plugins.get( 'ContextualBalloon' );
+
+		contextualBalloon.on<ContextualBalloonGetPositionOptionsEvent>( 'getPositionOptions', evt => {
+			const position = evt.return;
+
+			if ( !position || !stickyPanel.isSticky || !stickyPanel.element ) {
+				return;
+			}
+
+			// Measure toolbar (and menu bar) height.
+			const stickyPanelHeight = new Rect( stickyPanel.element ).height;
+
+			// Handle edge case when the target element is larger than the limiter.
+			// It's an issue because the contextual balloon can overlap top table cells when the table is larger than the viewport
+			// and it's placed at the top of the editor. It's better to overlap toolbar in that situation.
+			// Check this issue: https://github.com/ckeditor/ckeditor5/issues/15744
+			const target = isFunction( position.target ) ? position.target() : position.target;
+			const limiter = isFunction( position.limiter ) ? position.limiter() : position.limiter;
+
+			if ( target && limiter && new Rect( target ).height >= new Rect( limiter ).height - stickyPanelHeight ) {
+				return;
+			}
+
+			// Ensure that viewport offset is present, it can be undefined according to the typing.
+			const viewportOffsetConfig = { ...position.viewportOffsetConfig };
+			const newTopViewportOffset = Math.max( stickyPanelHeight, viewportOffsetConfig.top || 0 );
```

Comment:
I believe this should be a sum of those 2 offsets. Otherwise, we would have such an effect on our docs:

![Screenshot 2024-08-02 at 15 08 57](https://github.com/user-attachments/assets/0ac80678-2469-4c33-8680-e6e296686664)

The sticky pannel is offset from the window top by the size of viewport offset so contextual balloon should be offset by the sum of those.

Reply:
Yep, you are right. I'll check that.